crates/core/component/ibc/src/component/msg_handler/upgrade_client.rs

```diff
@@ -42,10 +44,34 @@ impl MsgHandler for MsgUpgradeClient {
 
         let upgraded_client_state_tm = TendermintClientState::try_from(self.client_state.clone())
             .context("client state is not a Tendermint client state")?;
+
         let upgraded_consensus_state_tm =
             TendermintConsensusState::try_from(self.consensus_state.clone())
                 .context("consensus state is not a Tendermint consensus state")?;
 
+        // this part of the IBC spec is sketchy and undocumented.
+        // we're going to reset some of the upgraded client state fields to their default values before verifying
+        // the remote's inclusion of the client state; this is what ibc-go does.
+        //
+        // note that later on in this function, we only allow the fields that we didn't reset here to change across upgrades.
+        // this means that there's no danger of a relayer sending an updated state
+        // with a new trust level and lying to us here, causing us to reconfigure the client with a bogus trust level.
+        //
+        // relevant ibc-go code:
+        // https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/upgrade.go#L74
+        // https://github.com/cosmos/ibc-go/blob/2555a7c504a904064d659e4c1a3a74000887f73d/modules/core/02-client/keeper/keeper.go#L552-L564
```

Comment:
Interesting. I realize we haven't tested counterparty host chain upgrade ever, maybe we should do that.